src/rpc/net.cpp

```diff
@@ -278,6 +278,7 @@ static RPCHelpMan addnode()
                 {
                     {"node", RPCArg::Type::STR, RPCArg::Optional::NO, "The node (see getpeerinfo for nodes)"},
                     {"command", RPCArg::Type::STR, RPCArg::Optional::NO, "'add' to add a node to the list, 'remove' to remove a node from the list, 'onetry' to try a connection to the node once"},
+                    {"block_relay_only", RPCArg::Type::BOOL, RPCArg::Default{false}, "If set, treat as block-relay-only connection (deactivating transaction and addr relay on this connection)."},
```

Comment:
This should be a `connection_type` String param, not a Boolean (see also #20551).

(Throw an exception if an unsupported value is passed, for this PR)

Reply:
Done with the latest push.